pkg/ingester/ingester_v2.go

```diff
@@ -135,7 +136,10 @@ func (i *Ingester) v2Push(ctx old_ctx.Context, req *client.WriteRequest) (*clien
 			// 400 error to the client. The client (Prometheus) will not retry on 400, and
 			// we actually ingested all samples which haven't failed.
 			if err == tsdb.ErrOutOfBounds || err == tsdb.ErrOutOfOrderSample || err == tsdb.ErrAmendSample {
-				lastPartialErr = err
+				if firstPartialErr == nil {
+					firstPartialErr = errors.Wrapf(err, "series=%s", lset.String())
```

Comment:
This seems like premature optimisation. Do we know if this is indeed a problem? I would think that over the other allocs and network calls, this won't even show up.

Reply:
I see your point. Let me show you a different angle. We need to keep 1 error, either we keep the first one or last one doesn't make any difference (and practically same code). Keeping the first one saves few cycles in case there are several partial errors, so look a low hanging fruit we can get.

Reply:
Could the last error also be a follow up of the first error at times? If yes, another reason to return the first error.

Reply:
Good point @codesome. I also thought about it, and my conclusion has been that these errors should not depend on a previous error, but I can't guarantee it. Generally speaking, I believe the first error is more significative than the last one.